pkg/ceph/osd/agent.go

```diff
@@ -565,6 +565,13 @@ func refreshDeviceInfo(name string, nameToUUID map[string]string, scheme *PerfSc
 func (a *OsdAgent) startOSD(context *clusterd.Context, config *osdConfig) error {
 	newOSD := false
 	config.rootPath = path.Join(config.configRoot, fmt.Sprintf("osd%d", config.id))
+
+	if isFilestoreDevice(config) {
+		if err := remountFilestoreDeviceIfNeeded(context, config); err != nil {
```

Comment:
we actually don't know if it's a new OSD or not until after we do the remounting (if it's needed).  we check if it's a new OSD or not by the existence of its `whoami` file, which doesn't show up until after the filestore device is remounted.

I'm moving the declaration of the `newOSD` variable until after this check to make it more clear that we don't know if it's new or not yet.